api/providers/WarehouseProvider.cs

```diff
@@ -98,5 +139,39 @@ public List<Location> GetLocationsByWarehouseId(Guid warehouseId)
 
         return locationsOfspecificWarehouse;
     }
+    private void validateWarehouse(Warehouse warehouse)
+    {
 
+        var validationResult = _warehouseValidator.Validate(warehouse);
+        if (validationResult.IsValid == false)
+        {
+            throw new ValidationException(validationResult.Errors);
+        }
+    }
+
+}
+
+
+public class WarehouseValidator : AbstractValidator<Warehouse>
+{
+    public WarehouseValidator()
+    {
+        // juiste regels toepassen in de validator
+        RuleFor(Warehouse => Warehouse.Name)
+             .NotNull().WithMessage("warehouse name is required.")
+             .NotEmpty().WithMessage("Warehouse name cannot be empty");
+
+        RuleFor(Warehouse => Warehouse.Code)
+            .NotNull().WithMessage("warehouse code is required.")
+            .NotEmpty().WithMessage("Warehouse Code cannot be empty");
+
+        RuleFor(Warehouse => Warehouse.Contact)
+            .NotNull().WithMessage("warehouse contact is required.")
+            .NotEmpty().WithMessage("Warehouse contact cannot be empty.");
+
+        RuleFor(Warehouse => Warehouse.Address)
```

Comment:
Je mist nog validation in de validator als de waarde address.id wel echt bestaat in de database zie:
https://docs.fluentvalidation.net/en/latest/built-in-validators.html#predicate-validator